### ISSUE TITLE
"Code editor" mode: fix borders and placeholders color contrast

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#25562]
**Describe the bug**

As noted in the similar issue (#25539) for the "Edit as HTML" styles, also the "Code editor" mode isn't meant to be WYSIWYG thus there's no reason why the input fields (actually textareas) used in this mode should not meet accessibility guidelines for contrast color.

Worth reminding in "Code editor" mode both the title and the post content use native HTML textarea elements.

**1. The borders**
- Normal state: the textarea elements border color is `#ccc` which against a white background produces a contrast ratio of 1.61:1 while it should be at least 3:1 (non-text contrast).

<img width="1001" alt="Screenshot 2020-09-23 at 15 35 35" src="https://user-images.githubusercontent.com/1682452/94024886-34cecf80-fdb8-11ea-975d-d17f9cf9ba18.png">

- Focus state: the border color is `#1e1e1e` which is OK but since there's the need to make the border normal state color darker, this would likely need to be darkened as well.
- Focus state: I'd consider to make the focus style color blue as it's the case for all the other native textarea elements and inputs: not sure why it needs to be a dark grey. As said above, the purpose of this mode isn't to provide a WYSIWYG mode: these are native HTML textarea elements.

**2. The native placeholder text**
- For the post title, the color used for the native placeholder text is `rgba(30, 30, 30, 0.55)` which against a white background produces an insufficient contrast ratio of 3.79:1, see https://jdlsn.com/color/?type=rgb&color=30,%2030,%2030&color2=255,255,255&alpha=0.55. Since this is text, the contrast should be at least 4.5:1
- For the post content, the placeholder text color is `#72777c` which is OK but this is the color inherited from the core `forms.css`. I guess the block editor should use its own styling.

<img width="1003" alt="Screenshot 2020-09-23 at 15 42 21" src="https://user-images.githubusercontent.com/1682452/94024948-46b07280-fdb8-11ea-914d-41c9335ecc0b.png">

Worth noting that core makes sure that all native placeholder text have sufficient color contrast. I'd expect Gutenberg to follow this core best practice. See:

https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/css/forms.css#L228-L239

(Firefox also needs the opacity to be reset to 1)

**To reproduce**
- create a new post: do not add any content
- switch fo Code editor mode
- observe the insufficient color contrast of the borders and of the title placeholder

**Editor version (please complete the following information):**
- WordPress version: 5.6 trunk
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? both
- If the Gutenberg plugin is installed, which version is it? 9.0.0
